packages/strapi-hook-bookshelf/lib/relations.js

```diff
@@ -76,7 +76,7 @@ module.exports = {
     const response = await module.exports.findOne.call(this, params);
 
     // Only update fields which are on this document.
-    const values = params.parseRelationships === false ? params.values : Object.keys(removeUndefinedKeys(params.values)).reduce((acc, current) => {
+    const values = params.parseRelationships === false ? params.values : await Object.keys(removeUndefinedKeys(params.values)).reduce(async (acc, current) => {
```

Comment:
Putting the reduce function as an async function makes it uncatchable. 

Can you simply update L 243 to 255 to be executed after the 
`awaitPromise.all(relationsUpdates)`on line 258

Reply:
Sure, but I think it's better then this fix be applied by someone who has a greater knowledge of the code of strapi than me :)

Reply:
No problem

I'll close this PR and do the fix then ;) 

Thanks for contributing anyway :D